module-apps/apps-common/widgets/TimeSetSpinnerVertical.hpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#pragma once
+
+#include <widgets/spinners/Spinners.hpp>
+#include <gui/widgets/Style.hpp>
+#include <gui/widgets/text/TextConstants.hpp>
+#include <time/time_locale.hpp>
+
+#include <string>
+
+namespace style::time_set_spinner_vertical
+{
+    namespace focus
+    {
+        inline constexpr auto size = 6U;
+    } // namespace focus
+
+    inline constexpr auto small_margin = 6U;
+    inline constexpr auto big_margin   = 44U;
+} // namespace style::time_set_spinner_vertical
+
+namespace gui
+{
+    /// Time set spinner widget class with option for dynamic switching between 24/12-hour format
+    /// Automatically recalculates hour upon switching format
+    /// Can be used as a basic time displaying widget when @ref EditMode set to 'Browse'
+    /// @ref KeyCode::KEY_ENTER switches hour -> minute -> time format
+    /// @ref KeyCode::KEY_RF switches back
+    /// Two time formats are supported:
+    /// utils::time::Locale::TimeFormat::FormatTime12H
+    /// utils::time::Locale::TimeFormat::FormatTime24H
+    class TimeSetSpinnerVertical : public VBox
+    {
+      public:
+        TimeSetSpinnerVertical(
```

Comment:
this class by design is combining view and model please at least move it to bell specific layouts :bow: